### PR TITLE
Prevent firing loadMore function multiple times

### LINKED ISSUE #143: loadmore twice
https://github.com/guillaumervls/react-infinite-scroll/issues/44

### PR DESCRIPTION
**Pull request by dominik-zeglen**
Fixes #143 as @ta-toshio suggested.

### CODE REVIEW

File: dist/InfiniteScroll.js [hunk 1]
```diff
@@ -95,7 +99,10 @@ var InfiniteScroll = (function(_Component) {
       )
     );
 
-    _this.scrollListener = _this.scrollListener.bind(_this);
+    _this.scrollListener = (0, _debounce2.default)(
+      _this.scrollListener.bind(_this),
+      100
```
Comment (joshuaobrien) [inline dist/InfiniteScroll.js]: What if it takes longer than this to resolve/reject?

**Comment by joshuaobrien** [comment]
I don't think this will actually fix the issue. Could there be a way to wait for the initial request to finish before the next is sent?

**Comment by dominik-zeglen** [comment]
I spent a few hours trying to figure that out and couldn't find any way doing it in a better way. The only thing we can do now is to pass debounce delay as a prop, so the client would adjust it manually. I think it's still better solution than not fixing it at all.
We can also merge this now and leave another issue to improve this solution later.

**Comment by joshuaobrien** [comment]
Will adding lodash as a dependency increase the component's size much?

**Comment by dominik-zeglen** [comment]
No, because debounce function is imported as a submodule, so it shouldn't be more than a few additional bytes.

**Comment by amir-beheshty** [comment]
I came across the same issue. Although this isn't a bulletproof solution I think it's definitely an improvement.

**Comment by joshuaobrien** [comment]
My main hesitation is adding an additional dependency, given there’s currently only one direct dependency.

**Comment by irfanalef2060** [comment]
I came across multiple times load more calling problem, for now, I have fixed using loadsh debounce method for deferring repeatedly calls. It would be better if we merge this pr so that others won't face this problem

**Pull request closed without merging**